src/slices/appSlice.ts

```diff
@@ -39,35 +39,38 @@ const initialState: AppState = {
     id: null,
     name: null,
   },
-  agents: []
+  agents: [],
 };
 
 export const loadPoolIds = createAsyncThunk(
-  'app/loadPoolIds',
+  "app/loadPoolIds",
   async (contractAddress: string) => {
     const contract = new ContractInteractor(contractAddress);
     return await contract.getPoolIds();
   }
 );
 
 export const loadAgent = createAsyncThunk(
-  'app/loadAgent',
-  async ({poolId, index}: {poolId: string, index: number}, thunkAPI) => {
+  "app/loadAgent",
+  async ({ poolId, index }: { poolId: string; index: number }, thunkAPI) => {
     // @ts-ignore
     const state: RootState = thunkAPI.getState();
-    const contract = new ContractInteractor(state.app.environment.contractAddress);
+    const contract = new ContractInteractor(
+      state.app.environment.contractAddress
+    );
     const agent = await contract.getAgentAt(poolId, index);
-    const {manifest} = await fetch(`${network.ipfs}/${agent[1]}`)
-      .then(response => {
+    const { manifest } = await fetch(`${network.ipfs}/${agent[3]}`).then(
```

Comment:
Interesting, does the `getAgentAt` returns different arguments for these new contracts?